### PR TITLE
Use human-friendly format for namui bundle index file

### PR DESCRIPTION
**Pull request by bigfoodK**
# What I did
Previously, we saved it in resources property of namui.config.json.
- But it's annoying to modify json directly.
- And we just need resources in config for now

So
- Use custom format that is human-friendly. It looks like .gitignore
- Remove `namui.config.json` and use `.namuibundle`.

# `.namuibundle` format
```
# comment, this sould be ignored

path/to/src
path/to/src:path/to/dest
```
- One item should be just one line
- Line starts with # will be ignored
- Empty line will be ignored

### CODE REVIEW

**Comment by namse** [comment]
I think the word `index` confuse me because I usually use `index` for for-loop... i guess?
What about use `list` or plural? Index doesn't feel that there are many data, but just feel single data.

**Comment by bigfoodK** [comment]
``# New features
## **: Test all directory below
`dir/**/file` on
```
dir/
├── dir1/
│   ├── file
│   └── dir3/
│       └── file
└── dir2/
    └── file
```
matches
- dir/dir1/file
- dir/dir2/file
- dir/dir1/dir3/file

## *: Same as regex's
`dir/file*` on 
```
dir/
├── file1
├── file2
└── 3file
```
matches
- dir/file1
- dir/file2

## !: exclude files
```
dir
!dir/dir2
```
on
```
dir/
├── dir1/
│   └── file1
└── dir2/
    └── file2
```
matches
- dir/dir1/file1

## #: comment

```
#path/to/file
#!path/to
```
nothing will happen

File: namui-cli/src/util/get_namui_bundle_list/get_namui_bundle_list.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+use super::{
```
Comment (namse) [inline namui-cli/src/util/get_namui_bundle_list/get_namui_bundle_list.rs:1]: I think directory name should be bundle_list, not get_namui_bundle_list..?

`get_namui_bundle_list/get_namui_bundle_list.rs` --> `bundle_list/get_namui_bundle_list`

File: namui-cli/src/util/get_namui_bundle_list/lexicon/namui_bundle_list.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+use super::{ExcludeOperation, IncludeOperation};
+use std::{collections::HashMap, path::PathBuf};
+
+type SrcDestPathMap = HashMap<PathBuf, PathBuf>;
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct NamuiBundleList {
```
Comment (namse) [inline namui-cli/src/util/get_namui_bundle_list/lexicon/namui_bundle_list.rs:7]: I suggest rename it again... to "NamuiBundleMenifest" or something like that. How about this?
It's not actually the list of bundle, it is kind of information what kind of bundle namui-cli should take. So it should have prefer name for the purpose.

File: namui-cli/src/util/get_namui_bundle_list/lexicon/namui_bundle_list.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    include: Vec<IncludeOperation>,
+    exclude: Vec<ExcludeOperation>,
+}
+impl NamuiBundleList {
+    pub fn new(include: Vec<IncludeOperation>, exclude: Vec<ExcludeOperation>) -> Self {
+        Self { include, exclude }
+    }
+
+    pub fn flatten(
```
Comment (namse) [inline namui-cli/src/util/get_namui_bundle_list/lexicon/namui_bundle_list.rs:16]: I think naming is not good... `query`?

**Review by namse — CHANGES_REQUESTED** [review]

File: namui-cli/src/util/get_namui_bundle_list/lexicon/namui_bundle_list.rs (hunk 2 above)
Comment (bigfoodK) [reply namui-cli/src/util/get_namui_bundle_list/lexicon/namui_bundle_list.rs:7]: Aaah that looks good

File: namui-cli/src/util/namui_bundle_manifest/get_namui_bundle_manifest.rs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    lexicon::{Lexer, NamuiBundleManifest},
+    token::Tokenizer,
+};
+use std::{fs, path::PathBuf};
+
+pub fn get_namui_bundle_manifest(
+    project_root_path: &PathBuf,
+) -> Result<NamuiBundleManifest, String> {
+    let namui_bundle_list_path = project_root_path.join(".namuibundle");
+    let namui_bundle_list_string = match namui_bundle_list_path.exists() {
```
Comment (namse) [inline namui-cli/src/util/namui_bundle_manifest/get_namui_bundle_manifest.rs:11]: Here rename too?

**Review by namse — CHANGES_REQUESTED** [review]

File: namui-cli/src/util/namui_bundle_manifest/get_namui_bundle_manifest.rs (hunk 4 above)
Comment (bigfoodK) [reply namui-cli/src/util/namui_bundle_manifest/get_namui_bundle_manifest.rs]: HAWK EYE

**Review by namse — APPROVED** [review]

**Pull request merged**